File: app/android/app/src/main/java/com/gonezo/multiplatform/audioextraction/infrastructure/llm/DefaultLlmGuard.java
package com.gonezo.multiplatform.audioextraction.infrastructure.llm;

import com.gonezo.multiplatform.audioextraction.domain.error.AudioExtractionException;
import com.gonezo.multiplatform.audioextraction.domain.error.ErrorCode;
import com.gonezo.multiplatform.audioextraction.domain.model.Transcript;

public final class DefaultLlmGuard implements LlmGuard {
  @Override
  public void validate(Transcript transcript, LlmConfig config) {
    String text = transcript == null || transcript.text() == null ? "" : transcript.text();
    if (text.isBlank()) {
      throw new AudioExtractionException(ErrorCode.TRANSCRIPTION_FAILED, "Transcript is empty");
    }
    if (text.length() > config.maxInputChars()) {
      throw new AudioExtractionException(ErrorCode.POLICY_REJECTED, "Transcript exceeds maxInputChars");
    }
  }
}

File: core/src/main/kotlin/com/gonezo/audioextraction/infrastructure/llm/DefaultLlmGuard.kt
package com.gonezo.audioextraction.infrastructure.llm

import com.gonezo.audioextraction.domain.error.AudioExtractionException
import com.gonezo.audioextraction.domain.error.ErrorCode
import com.gonezo.audioextraction.domain.model.Transcript

class DefaultLlmGuard : LlmGuard {
  override fun validate(transcript: Transcript, config: LlmConfig) {
    val text = transcript.text
    if (text.isBlank()) {
      throw AudioExtractionException(ErrorCode.TRANSCRIPTION_FAILED, "Transcript is empty")
    }
    if (text.length > config.maxInputChars) {
      throw AudioExtractionException(ErrorCode.POLICY_REJECTED, "Transcript exceeds maxInputChars")
    }
  }
}
